strictNullChecks thinks ambient declarations are unassigned and complains when they're used

**TypeScript Version:** 

https://github.com/Microsoft/TypeScript/commit/3853bb86d0f26e150666ab736092824eb781babc (master after merging #7140 )

**Code**

```ts
declare const foo: string;
console.log(foo);
```

`tsc -t es5 --strictNullChecks ./foo.ts`

**Expected behavior:**

Compiles.

**Actual behavior:** 

`foo.ts(2,13): error TS2454: Variable 'foo' is used before being assigned.`

---

Allowing `foo` to be undefined makes it compile, but this shouldn't be required of course.

```typescript
declare const foo: string | undefined;
console.log(foo);
```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -7725,7 +7725,7 @@ namespace ts {
                 return;
             }
             const declaration = symbol.valueDeclaration;
-            if (!declaration || declaration.kind !== SyntaxKind.VariableDeclaration || (<VariableDeclaration>declaration).initializer) {
+            if (!declaration || declaration.kind !== SyntaxKind.VariableDeclaration || (<VariableDeclaration>declaration).initializer || isInAmbientContext(declaration)) {
                 return;
             }
             const parentParentKind = declaration.parent.parent.kind;

